### PR TITLE
perf(snuba): Properly trace snuba queries when we're using a thread pool.

### PR DESCRIPTION
**Pull request by wedamija**
Since we're running batch snuba queries in a thread pool, by default the sentry_sdk uses a separate
hub for each thread, which causes the individual threaded traces to end up in a separate event. To
fix this, @untitaker suggested that for each task we clone `Hub.current` and pass it to the task, then
use that to create the threaded spans. This works well, and correctly groups the threaded tasks
under the outer span that creates the tasks.

### CODE REVIEW

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Review by untitaker — APPROVED** [review]
Yeah this works. If for whatever reason you want to set an event tag in a worker thread, and have that tag appear in crashes happening after the threadpool invocation, use `Hub.current` instead of `Hub(Hub.current)`

**Comment by dashed** [comment]
@untitaker what's the difference between `Hub(Hub.current)` and `Hub.current`?

**Comment by untitaker** [comment]
@dashed the difference is that the tags dictionary (and other scope/context data) is shared across threads, so all threads see tags from other threads and after your threadpool call the tags from the threads will "leak" into the threadpool-caller's scope. That last part may be useful. If you set tags right now in your function, they are rarely visible in events.

File: src/sentry/utils/snuba.py [hunk 1]
```diff
@@ -556,12 +557,20 @@ def snuba_query(params):
...
             raise SnubaError(err)
 
-    if len(snuba_param_list) > 1:
-        query_results = _query_thread_pool.map(snuba_query, query_param_list)
-    else:
-        # No need to submit to the thread pool if we're just performing a
-        # single query
-        query_results = [snuba_query(query_param_list[0])]
+    with sentry_sdk.start_span(
+        op="start_snuba_query", description="running {} snuba queries".format(len(snuba_param_list))
```
Comment (markstory) [inline src/sentry/utils/snuba.py]: Any chance of unicode bytes being in your param list?

Comment (wedamija) [reply src/sentry/utils/snuba.py]: Maybe, but this is just taking the `len` of the list. Regardless, we should likely use `u` here.

**Pull request merged**